src/elfpy/pricing_models.py

```diff
@@ -539,6 +542,180 @@ class HyperdrivePricingModel(PricingModel):
+        stretched_time_remaining,
+    ):
+        assert (
+            base_asset_in > 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected base_asset_in > 0, not {base_asset_in}!"
+        assert (
+            share_reserves >= 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected share_reserves >= 0, not {share_reserves}!"
+        assert (
+            bond_reserves >= 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected bond_reserves >= 0, not {bond_reserves}!"
+        assert (
+            share_buffer >= 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected share_buffer >= 0, not {share_buffer}!"
+        assert (
+            liquidity_pool >= 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected liquidity_pool >= 0, not {liquidity_pool}!"
+        assert (
+            rate >= 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected rate >= 0, not {rate}!"
+        assert (
+            1 > time_remaining >= 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected 1 > time_remaining >= 0, not {time_remaining}!"
+        assert (
+            stretched_time_remaining >= 0
+        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected stretched_time_remaining >= 0, not {stretched_time_remaining}!"
+        assert share_price >= init_share_price >= 1, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: expected share_price >= init_share_price >= 1, not"
+            f" share_price={share_price} and init_share_price={init_share_price}!"
+        )
+        r"""
+        Computes the amount of LP tokens to be minted for a given amount of base asset
+
+        .. math::
+
+        y = \frac{(z - \Delta z)(\mu \cdot (\frac{1}{1 + r \cdot t(d)})^{\frac{1}{\tau(d_b)}} - c)}{2}
+
+        """
+        if self.verbose:
+            print(f"  inputs: base_asset_in={base_asset_in}, share_reserves={share_reserves}, bond_reserves={bond_reserves}, share_buffer={share_buffer}, init_share_price={init_share_price}, share_price={share_price}, liquidity_pool={liquidity_pool}, rate={rate}, time_remaining={time_remaining}, stretched_time_remaining={stretched_time_remaining}")
+        d_share_reserves = base_asset_in / share_price
+        if self.verbose:
+            print(f"  d_share_reserves={d_share_reserves} (base_asset_in / share_price = {base_asset_in} / {share_price})")
+        if share_reserves > 0:  # normal case where we have some share reserves
+            lp_out = d_share_reserves * liquidity_pool / (share_reserves - share_buffer)
+        else:  # initial case where we have 0 share reserves
+            lp_out = d_share_reserves
+        d_token_reserves = (share_reserves + d_share_reserves) / 2 * (
+            init_share_price * (1 + rate * time_remaining) ** (1 / stretched_time_remaining) - share_price
+        ) - bond_reserves
+        if self.verbose:
```

Comment:
can you combine all of the verbose prints per function? and remove debugging prints